pkg/scheduler/framework/plugins/interpodaffinity/plugin_test.go

```diff
@@ -0,0 +1,176 @@
+		expectedHint   framework.QueueingHint
+		expectedErr    bool
+	}{
+		{
+			name:         "add a pod with pod affinity which matches the terms",
+			pod:          st.MakePod().Name("p").PodAffinityIn("service", "region", []string{"securityscan", "value2"}, st.PodAffinityWithRequiredReq).Obj(),
+			newPod:       st.MakePod().Node("fake-node").Label("service", "securityscan").Obj(),
+			expectedHint: framework.Queue,
+		},
+		{
+			name:         "add an un-scheduled pod",
+			pod:          st.MakePod().Name("p").PodAffinityIn("service", "region", []string{"securityscan", "value2"}, st.PodAffinityWithRequiredReq).Obj(),
+			newPod:       st.MakePod().Label("service", "securityscan").Obj(),
+			expectedHint: framework.QueueSkip,
+		},
+		{
+			name:         "add a pod with pod affinity which doesn't match the terms",
+			pod:          st.MakePod().Name("p").PodAffinityIn("service", "region", []string{"securityscan", "value2"}, st.PodAffinityWithRequiredReq).Obj(),
+			newPod:       st.MakePod().Node("fake-node").Label("aaa", "a").Obj(),
+			expectedHint: framework.QueueSkip,
+		},
+		{
+			name:         "update a pod with pod affinity which matches the terms",
+			pod:          st.MakePod().Name("p").PodAffinityIn("service", "region", []string{"securityscan", "value2"}, st.PodAffinityWithRequiredReq).Obj(),
+			newPod:       st.MakePod().Node("fake-node").Label("service", "securityscan").Obj(),
+			oldPod:       st.MakePod().Node("fake-node").Label("aaa", "a").Obj(),
+			expectedHint: framework.Queue,
+		},
+		{
+			name:         "update an un-scheduled pod",
+			pod:          st.MakePod().Name("p").PodAffinityIn("service", "region", []string{"securityscan", "value2"}, st.PodAffinityWithRequiredReq).Obj(),
+			newPod:       st.MakePod().Label("service", "securityscan").Obj(),
+			oldPod:       st.MakePod().Label("service", "securityscan").Obj(),
+			expectedHint: framework.QueueSkip,
+		},
+		{
+			name:         "update a pod with pod affinity which doesn't match the terms",
+			pod:          st.MakePod().Name("p").PodAffinityIn("service", "region", []string{"securityscan", "value2"}, st.PodAffinityWithRequiredReq).Obj(),
+			newPod:       st.MakePod().Node("fake-node").Label("aaa", "a").Obj(),
+			oldPod:       st.MakePod().Node("fake-node").Label("service", "securityscan").Obj(),
+			expectedHint: framework.QueueSkip,
+		},
+		{
+			name: "update a pod which doesn't match pod's affinity - multiple terms case",
+			pod: st.MakePod().Name("p").PodAffinityExists("aaa", "hostname", st.PodAffinityWithRequiredReq).
+				PodAffinityExists("bbb", "hostname", st.PodAffinityWithRequiredReq).PodAffinityExists("aaa", "hostname", st.PodAffinityWithRequiredReq).Obj(),
+			newPod:       st.MakePod().Node("fake-node").Label("service", "securityscan").Obj(),
+			oldPod:       st.MakePod().Node("fake-node").Label("aaa", "a").Obj(),
+			expectedHint: framework.QueueSkip,
+		},
+		{
+			name: "update a pod which matches pod's affinity - multiple terms case",
```

Comment:
Can you add cases for updating an existing pod's label, which matches /doesn't matches pod's pod-anti-affinity?